packages/block-library/src/audio/edit.native.js

```diff
@@ -67,11 +68,17 @@ function AudioEdit( {
 		};
 	}
 
-	function onSelectURL() {
-		// TODO: Set up add audio from URL flow
+	function onSelectURL( newSrc ) {
+		if ( newSrc !== src ) {
+			if ( isURL( newSrc ) ) {
+				setAttributes( { src: newSrc, id: undefined } );
+			} else {
+				createErrorNotice( __( 'Invalid URL. Audio file not found.' ) );
+			}
+		}
 	}
 
-	function onUploadError( message ) {
```

Comment:
@ceyhun I had utilized this functionality for showing the URL validation error because I saw it here but when I tested the behavior, I was not able to get it to work. So I am wondering if there's something special I need to do. Let me know if you have any thoughts when you get a chance to check this PR out.

Reply:
Seems like `noticeUI` passed into `MediaPlaceholder` needs to be rendered in there for it to work like [this](https://github.com/WordPress/gutenberg/blob/467e2068d88797e3e408021778989b0cccf9bcac/packages/components/src/higher-order/with-notices/README.md#usage). Which should render the `NoticeList` component according to [its implementation](https://github.com/WordPress/gutenberg/blob/467e2068d88797e3e408021778989b0cccf9bcac/packages/components/src/higher-order/with-notices/index.js#L77-L81) and `NoticeList` seems to already have [a native implementation](https://github.com/WordPress/gutenberg/blob/467e2068d88797e3e408021778989b0cccf9bcac/packages/components/src/notice/index.native.js).

I think these parts of the Audio block were copied over directly from [web's edit.js](https://github.com/WordPress/gutenberg/blob/654eb21a136aa2e076b71f6b80253ed02e2d01a1/packages/block-library/src/audio/edit.js#L93-L96) when the native edit file was first created and we didn't see it functioning on native yet.

Reply:
> Seems like noticeUI passed into MediaPlaceholder needs to be rendered in there for it to work like this. Which should render the NoticeList component according to its implementation and NoticeList seems to already has a native implementation.

Totally missed that it wasn't in the `render` function :) so once we add that there we should be good. 

> I think these parts of the Audio block were copied over directly from web's edit.js when the native edit file was first created and we didn't see it functioning on native yet.

Makes sense!

Reply:
I created a PR here https://github.com/WordPress/gutenberg/pull/30273 to resolve these notice issues. Once it is resolved and merged into trunk, I will sync the changes with this PR and it can be reviewed and merged in.